src/api/goods.ts

```
@@ -3,14 +3,35 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+    if (!response.ok) {
+      throw new Error(`HTTP Error: ${response.status}`);
+    }
+
+    return response.json();
+  } catch (error) {
+    return [];
```

This catch block swallows all errors and returns an empty array. That masks network or parsing failures so the UI/test code cannot detect errors. This violates the checklist requirement: "Catch and handle network or parsing errors during fetch. Propagate or display the error state in a way that can be verified in code (e.g., setting error state and conditional rendering) (see checklist item 3)." Instead of returning `[]`, rethrow the error (or return a rejected promise) so callers can set error state and render error UI.